File: bioskopina_mobile_/lib/models/post.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/comment.dart';
import '../models/user.dart';  // Import User model

part 'post.g.dart';

@JsonSerializable()
class Post {
  int? id;
  int? userId;
  String? content;
  int? likesCount;
  int? dislikesCount;
  DateTime? datePosted;
  List<Comment>? comments;
  User? user; // NEW: Embedded User object

  Post(
    this.id,
    this.userId,
    this.content,
    this.likesCount,
    this.dislikesCount,
    this.datePosted,
    this.comments,
    this.user, // NEW
  );

  factory Post.fromJson(Map<String, dynamic> json) => _$PostFromJson(json);
  Map<String, dynamic> toJson() => _$PostToJson(this);
}

File: backend/Bioskopina.Model/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Bioskopina.Model
{
    public class Post
    {
        public int Id { get; set; }

      

        public int UserId { get; set; }

        public string Content { get; set; } = null!;

        public int LikesCount { get; set; }

        public int DislikesCount { get; set; }

        public DateTime DatePosted { get; set; }

     
        
        public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

        public virtual User User { get; set; } = null!;
    }
}
